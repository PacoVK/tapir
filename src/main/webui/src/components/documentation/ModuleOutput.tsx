import React from "react";
import { Box, Stack, Typography } from "@mui/material";
import { ModuleOutputProps } from "../../types";
import Container from "@mui/material/Container";

const ModuleOutput = (props: ModuleOutputProps) => {
  const { outputs } = props;
  return (
    <Stack justifyContent="center" alignItems="left" spacing={2}>
      {outputs && outputs.length > 0 ? (
        outputs.map((output, index) => {
          return (
            <Box key={`module-output-${index}`}>
              <Typography>
                <strong>{output.name}</strong>
              </Typography>
              <Container maxWidth="sm">
                <Typography>{output.description}</Typography>
              </Container>
            </Box>
          );
        })
      ) : (
        <Typography>This module has no outputs</Typography>
      )}
    </Stack>
  );
};

export default ModuleOutput;
