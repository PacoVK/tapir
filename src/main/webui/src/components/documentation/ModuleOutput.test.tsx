import { render } from "@testing-library/react";
import ModuleOutput from "./ModuleOutput";

describe("<ModuleOutput /> spec", () => {
  it("renders the ModuleOutput", () => {
    const view = render(
      <ModuleOutput
        outputs={[{ name: "github_roles" }, { name: "registries" }]}
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
