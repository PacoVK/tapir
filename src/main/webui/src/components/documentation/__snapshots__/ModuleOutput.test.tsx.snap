// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleOutput /> spec renders the ModuleOutput 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-c08hs3-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            <strong>
              github_roles
            </strong>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            <strong>
              registries
            </strong>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-c08hs3-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          <strong>
            github_roles
          </strong>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          />
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          <strong>
            registries
          </strong>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-zzzaw2-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
